### PR TITLE
fix(node): handle offshoot promise rejections

### LINKED ISSUE #10367: Server crashes due to multiple runtime errors in Astro component (Astro + React)
### Astro Info

```block
Astro                    v4.4.15
Node                     v18.19.0
System                   macOS (x64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/node
Integrations             @astrojs/react
```

### If this issue only occurs in one browser, which browser is a problem?

All browsers

### Describe the Bug

When rendering an Astro component (`index.astro`), if there is an error in a React component preceded by another runtime error, it will result in a server crash.

Example:
```
---
// index.astro
import ReactComp from '../components/react-comp';

const foo = { bar: null } as unknown as { bar: { baz: string[] } };
---

<!-- React component has to error first -->
<ReactComp foo={foo} />
{ foo.bar.baz.length > 0 && (
  <div />
)}
```

### Steps to reproduce
1. Run `npm run build`
2. Run `npm run preview`
3. Go to localhost
4. Go to `/` page, server crashes.

### What's the expected result?

Should not crash the server no matter how the error happen.

### Link to Minimal Reproducible Example

https://github.com/frendyguo/astro-react-runtime-error-crash

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by lilnasy**
## Changes
- Fixes #10367 thoroughly but only for the node adapter.

## Testing
- Added

## Docs
- Does not affect usage.

### CODE REVIEW

File: packages/integrations/node/test/errors.test.js [hunk 1]
```diff
@@ -17,10 +22,30 @@ describe('Errors', () => {
 	let devPreview;
 
 	before(async () => {
-		devPreview = await fixture.preview();
+		// devPreview = await fixture.preview();
```
Comment (ematipico) [inline packages/integrations/node/test/errors.test.js:25]: To remove

**Review by ematipico — APPROVED** [review]

File: packages/integrations/node/test/errors.test.js (hunk 1 above)
Comment (lilnasy) [reply packages/integrations/node/test/errors.test.js:25]: It isnt being used yet

Comment (ematipico) [reply packages/integrations/node/test/errors.test.js:25]: What do you mean with yet? Can you be more precise, please?

Comment (lilnasy) [reply packages/integrations/node/test/errors.test.js:25]: The other two tests in the suite are skipped. The one I added starts a server within a worker. The preview server being started here is not used.

Comment (lilnasy) [reply packages/integrations/node/test/errors.test.js:25]: and it listens on the same port

Comment (ematipico) [reply packages/integrations/node/test/errors.test.js:25]: Usually, if you want to keep commented code around, it's customary to leave a comment explains that :)

Comment (lilnasy) [reply packages/integrations/node/test/errors.test.js:25]: Sure ill add that

Comment (lilnasy) [reply packages/integrations/node/test/errors.test.js:25]:
```suggestion
		// The two tests that need the server to run are skipped
		// devPreview = await fixture.preview();
```

**Review by bluwy — APPROVED** [review]
Still have a little concern of handling `unhandledRejection` for users by default, but I think it's worth trying it out.

**Comment by lilnasy** [comment]
> Still have a little concern of handling unhandledRejection for users by default, but I think it's worth trying it out.

That's fair. I think it makes sense for the adapter to do it because there is no natural place for the developer to add the listener - api route top-level, middleware top-level?

**Pull request merged**